python/data_structures/linked_list.py

```diff
@@ -34,15 +34,64 @@ def includes(self, value):
             current = current.next
         return False
 
+    def append(self, value):
+        new_node = Node(value)
+        if self.head:
+            current = self.head
+            while current.next:
+                current = current.next
+            current.next = new_node
+        else:
+            self.head = new_node
+        return new_node
+
+    def insert_before(self, value, new_value):
+        previous = current = self.head
+        if self.head is None:
+            raise TargetError
+        if self.includes(value) is False:
+            raise TargetError
```

Comment:
There's a case to be made that lines 50 and 51 here are unnecessary. Looking back at your .includes method, the condition on 52 should adequately cover a case where self.head is None.